common/service-loader/src/main/java/io/helidon/common/serviceloader/HelidonServiceLoader.java

```diff
@@ -0,0 +1,360 @@
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.annotation.Priority;
+
+import io.helidon.common.Prioritized;
+
+/**
+ * Helidon specific support for Java Service Loaders.
+ * <p>
+ * This service loader:
+ * <ul>
+ * <li>Can have additional implementations added</li>
+ * <li>Uses priorities defined either by {@link io.helidon.common.Prioritized}
+ * or by {@link javax.annotation.Priority}</li>
+ * <li>Can have exclusions defined by an exact implementation class name, either
+ * in {@link Builder#addExcludedClass(Class)} or {@link Builder#addExcludedClassName(String)} or
+ * by a system property {@value #SYSTEM_PROPERTY_EXCLUDE} that defines
+ * a comma separated list of fully qualified class names to be excluded.
+ * Note that if a class implements more than one service, it would be excluded from all.</li>
+ * </ul>
+ * <p>
+ * <i>Note on priority handling</i>
+ * <p>
+ * Service priority is defined by:
+ * <ul>
+ * <li>Value provided in {@link Builder#addService(Object, int)} (if used)</li>
+ * <li>then by {@link io.helidon.common.Prioritized#priority()} if service implements it</li>
+ * <li>then by {@link javax.annotation.Priority} annotation if present</li>
+ * <li>otherwise a default priority {@value #DEFAULT_PRIORITY} from {@link #DEFAULT_PRIORITY} is used</li>
+ * </ul>
+ * A service with lower priority number is returned before a service with a higher priority number.
+ * Services with the same priority have order defined by the order they are in the configured services
+ * and then as they are loaded from the {@link java.util.ServiceLoader}.
+ * Negative priorities are allowed.
+ * A service with priority {@code 1} will be returned before a service with priority {@code 2}.
+ * <p>
+ *
+ * @param <T> Type of the service to be loaded
+ * @see java.util.ServiceLoader
+ * @see #builder(java.util.ServiceLoader)
+ */
+public final class HelidonServiceLoader<T> implements Iterable<T> {
+    /**
+     * Default priority of a service.
+     */
+    public static final int DEFAULT_PRIORITY = 100;
+    /**
+     * System property used to exclude some implementation from the list of services that are configured for Java Service
+     * loader or services that are registered using {@link io.helidon.common.serviceloader.HelidonServiceLoader.Builder}.
+     */
+    public static final String SYSTEM_PROPERTY_EXCLUDE = "io.helidon.common.serviceloader.exclude";
+
+    private static final Logger LOGGER = Logger.getLogger(HelidonServiceLoader.class.getName());
+
+    private final List<T> services;
+
+    /**
+     * Create a builder for customizable service loader.
+     *
+     * @param javaServiceLoader the Java Service loader used to get service implementations
```

Comment:
I think you should remove the `java` prefix (everywhere); it is already clear that the parameter is the system service loader. This prefix seems both redundant and we probably don't want to set a precedent here ;-).

Reply:
Removing. I will replace it to `useSystemServiceLoader` when enabling/disabling it.

Reply:
Fixed